components/sync/driver/brave_sync_auth_manager.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/sync/driver/brave_sync_auth_manager.h"
+
+#include "base/base64.h"
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_sync/crypto/crypto.h"
+#include "brave/components/brave_sync/network_time_helper.h"
+
+namespace syncer {
+
+BraveSyncAuthManager::BraveSyncAuthManager(
+    signin::IdentityManager* identity_manager,
+    const AccountStateChangedCallback& account_state_changed,
+    const CredentialsChangedCallback& credentials_changed)
+    : SyncAuthManager(identity_manager,
+                      account_state_changed,
+                      credentials_changed) {}
+
+BraveSyncAuthManager::~BraveSyncAuthManager() {}
+
+void BraveSyncAuthManager::DeriveSigningKeys(const std::string& seed) {
+  VLOG(1) << __func__ << " seed=" << seed;
+  if (seed.empty())
+    return;
+  const std::vector<uint8_t> HKDF_SALT = {
+      72,  203, 156, 43,  64,  229, 225, 127, 214, 158, 50,  29,  130,
+      186, 182, 207, 6,   108, 47,  254, 245, 71,  198, 109, 44,  108,
+      32,  193, 221, 126, 119, 143, 112, 113, 87,  184, 239, 231, 230,
+      234, 28,  135, 54,  42,  9,   243, 39,  30,  179, 147, 194, 211,
+      212, 239, 225, 52,  192, 219, 145, 40,  95,  19,  142, 98};
+  std::vector<uint8_t> seed_bytes;
+  brave_sync::crypto::PassphraseToBytes32(seed, &seed_bytes);
+  brave_sync::crypto::DeriveSigningKeysFromSeed(seed_bytes, &HKDF_SALT,
```

Comment:
this is somewhat of a cryptography nit, but i believe you should be setting the info field in https://github.com/brave/brave-core/blob/cd5a08090094fc1a67033e2760c3a067b27e001d/components/brave_sync/crypto/crypto.cc#L46 to be an informative, non-empty value (like the string 'sync-auth-key') instead of hard-coding it in https://github.com/brave/brave-core/blob/cd5a08090094fc1a67033e2760c3a067b27e001d/components/brave_sync/crypto/crypto.cc#L46. that way, if we want to use the same seed to derive another key in the future, we just set the info field to something else.

Reply:
fine to keep this as-is though if it's necessary for backwards compatibility

Reply:
It was originally rewritten from sync v1 js code https://github.com/brave/crypto/blob/9f902f60f0f70cc753058b305854fea6be24982b/index.js#L126
so is the backward compatibility here means able to connect to sync v1 sync chain?
If yes, I think we can set meaningful info.

Reply:
right, we set it to 0 in v1 but later realized we should have set it to a meaningful string :).

Reply:
fixed in https://github.com/brave/brave-core/pull/5294/commits/e93b6cc04b460ba5570bf8c8230982a6c6f05fc3